velox/common/base/SimdUtil-inl.h

```diff
@@ -1436,4 +1436,189 @@ inline bool memEqualUnsafe(const void* x, const void* y, int32_t size) {
   return true;
 }
 
+namespace detail {
+template <typename T>
+T clearLeftmostSet(const T value) {
+  assert(value != 0);
+
+  return value & (value - 1);
+}
+
+template <typename T>
+unsigned FOLLY_ALWAYS_INLINE getFirstBitSet(const T value) {
+  assert(value != 0);
+
+  return __builtin_ctz(value);
+}
+
+template <>
+unsigned FOLLY_ALWAYS_INLINE getFirstBitSet<uint64_t>(const uint64_t value) {
+  assert(value != 0);
+
+  return __builtin_ctzl(value);
+}
+
+#if XSIMD_WITH_AVX2
+// AVX2 is faster than sse2
+#define SIMD_STRSTR
+using CharVector = xsimd::batch<uint8_t, xsimd::avx2>;
+#elif XSIMD_WITH_NEON
+#define SIMD_STRSTR
+using CharVector = xsimd::batch<uint8_t, xsimd::neon>;
+#endif
+
+#ifdef SIMD_STRSTR
+size_t FOLLY_ALWAYS_INLINE
+smidStrstrAnysize(const char* s, size_t n, const char* needle, size_t k) {
```

Comment:
This can be combined with `smidStrstrMemcmp`.  Just pass `k` as both template parameter and runtime parameter.